osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderHead.cs

```diff
@@ -56,12 +65,14 @@ protected override HitResult ResultFor(double timeOffset)
         {
             Debug.Assert(HitObject != null);
 
-            if (HitObject.JudgeAsNormalHitCircle)
-                return base.ResultFor(timeOffset);
-
-            // If not judged as a normal hitcircle, judge as a slider tick instead. This is the classic osu!stable scoring.
-            var result = base.ResultFor(timeOffset);
-            return result.IsHit() ? HitResult.LargeTickHit : HitResult.LargeTickMiss;
+            return HitObject.ClassicSliderBehaviour
+                // In classic slider behaviour, heads are considered fully hit if in the largest hit window.
+                // We can't award a full Great because the true Great judgement is awarded on the Slider itself,
+                // reduced based on number of ticks hit.
+                //
+                // so we use the most suitable LargeTick judgement here instead.
+                ? base.ResultFor(timeOffset).IsHit() ? HitResult.LargeTickHit : HitResult.LargeTickMiss
+                : base.ResultFor(timeOffset);
```

Comment:
Can we not do nested ternaries? They're just not very readable as far as I'm concerned.